src/webgpu/api/validation/encoding/encoder_state.spec.ts

```diff
@@ -74,9 +77,12 @@ g.test('pass_end_invalid_order')
       passes[index].end();
```

Comment:
Oh, after reviewing I thought of something. If you have already ended the pass, then ending it again should generate a validation error. This isn't implemented yet in Dawn.
https://bugs.chromium.org/p/dawn/issues/detail?id=1578
Sorry that I forgot about this when you were first implementing this test.

In fact, there are a number of other subtleties that I think mismatch even the original test before this PR. https://gpuweb.github.io/gpuweb/#dom-gpucommandencoder-begincomputepass

While looking at this I also just noticed an error in the spec. It appears to have an invalid state assertion. So there is still some stuff to iron out here.

Reply:
I just saw I already commented about this in https://github.com/gpuweb/cts/pull/1917#pullrequestreview-1144383146
Deja vu. In any case, let me review that too.